### PR TITLE
Passing Temporal Model Handling to User & Using Most-Recent Off Policy Carry Value During Rollouts

### PR DESCRIPTION
**Pull request by michael-lutz**
And some small rafactorings / renamings.

Using the most recent off-policy carry value seems to help in termination length when you get to large episode lengths. Doesn't seem to improve the per-rollout reward significantly. Going to leave as toggleable. Light blue is with off-policy carries, dark blue uses on-policy carries.

Screen Shot 2025-04-08 at 7 38 06 PM

### CODE REVIEW

File: examples/walking_rnn.py [hunk 1]
```diff
@@ -313,27 +313,44 @@ def run_critic(
...
+                carry=critic_carry,
+            )
+
+            transition_ppo_variables = ksim.PPOVariables(
+                log_probs=log_probs,
+                values=value.squeeze(-1),
+            )
+
+            initial_carry = self.get_initial_model_carry(rng)
+            next_carry = jax.tree.map(lambda x, y: jnp.where(transition.done, x, y), initial_carry, actor_critic_carry)
```
Comment (codekansas) [inline examples/walking_rnn.py]: yea, i think it is better to have this logic here than upstream. definitely more readable, easier to implement transformers as well

File: examples/walking.py [hunk 2]
```diff
@@ -612,7 +618,7 @@ def sample_action(
     # On MacOS or other devices with less memory, you can change the number
     # of environments and batch size to reduce memory usage. Here's an example
     # from the command line:
-    #   python -m examples.walking num_envs=8 batch_size=4
+    #   python -m examples.walking num_envs=8 rollouts_per_batch=4
```
Comment (codekansas) [inline examples/walking.py:558]: did you change back to `batch_size`?

File: ksim/task/ppo.py [hunk 3]
```diff
@@ -174,6 +173,16 @@ def compute_ppo_loss(
...
+    # The following should not have any singleton dimensions.
+    chex.assert_rank(on_policy_variables.values, 1)
+    chex.assert_rank(off_policy_variables.values, 1)
+    chex.assert_rank(ppo_inputs.advantages_t, 1)
+    chex.assert_rank(ppo_inputs.value_targets_t, 1)
+    chex.assert_rank(dones_t, 1)
+
+    # Log probs should have an extra dimension for the number of actions.
+    chex.assert_rank(on_policy_variables.log_probs, 2)
+    chex.assert_rank(off_policy_variables.log_probs, 2)
```
Comment (codekansas) [inline ksim/task/ppo.py:185]: nice

File: ksim/task/rl.py [hunk 4]
```diff
@@ -321,7 +325,7 @@ class RLConfig(xax.Config):
         help="Log the full trajectory every N steps.",
     )
     log_full_trajectory_every_n_seconds: float | None = xax.field(
-        60.0 * 10.0,
+        60.0,
```
Comment (codekansas) [inline ksim/task/rl.py]: why this new value?

**Review by codekansas — APPROVED** [review]
discussed offline

File: ksim/task/rl.py (hunk 4 above)
Comment (codekansas) [reply ksim/task/rl.py]: moved logging to separate thread
